src/daemon/context.rs

```diff
@@ -64,6 +65,9 @@ impl DaemonContext {
             local_rdht_shard: CacheBuilder::default()
                 .time_to_idle(Duration::from_secs(60 * 60))
                 .build(),
+            rdht_cache: CacheBuilder::default()
+                .time_to_idle(Duration::from_secs(60 * 30))
```

Comment:
This should be `time_to_live`, so that outdated entries are not kept alive simply by looking them up often.

We should also use a much shorter timeout (say 60 seconds) to minimize downtime when havens switch rendezvous points.